packages/terra-filter-view/src/_Frame.jsx

```diff
@@ -0,0 +1,540 @@
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleMouseDown = this.handleMouseDown.bind(this);
+    this.handleInputMouseDown = this.handleInputMouseDown.bind(this);
+  }
+
+  componentDidUpdate(previousProps, previousState) {
+    if (this.props.variant === Variants.DROPDOWN && Util.shouldPositionDropdown(previousState, this.state, this.dropdown)) {
+      clearTimeout(this.debounceTimer);
+      this.debounceTimer = setTimeout(this.positionDropdown, !previousState.isOpen ? 0 : 100);
+    }
+  }
+
+  componentWillUnmount() {
+    clearTimeout(this.debounceTimer);
+  }
+
+  setInput(input) {
+    this.input = input;
+  }
+
+  getDisplay() {
+    const { searchValue } = this.state;
+    const { disabled, placeholder } = this.props;
+
+    const inputAttrs = {
+      disabled,
+      placeholder,
+      ref: this.setInput,
+      onChange: this.handleSearch,
+      onMouseDown: this.handleInputMouseDown,
+      'aria-label': 'search',
+      className: cx('search-input', { 'right-padding': this.state.isFocused && this.state.searchValue }),
+    };
+
+    return <input {...inputAttrs} value={searchValue} />;
+  }
+
+  /**
+   * Clears the search-bar content.
+   */
+  clearSearch() {
+    if (this.state.searchValue !== '') {
+      this.setState({ searchValue: '', excludeBlur: false });
+    }
+    // Refocus input after clearing
+    this.input.focus();
+  }
+
+  /**
+   * Closes the dropdown.
+   */
+  closeDropdown() {
+    if (this.props.variant !== Variants.LIST) {
+      this.setState({
+        isAbove: false,
+        isFocused: document.activeElement === this.input || document.activeElement === this.select,
+        isOpen: false,
+        isPositioned: false,
+      });
+    } else {
+      this.setState({ isFocused: document.activeElement === this.input || document.activeElement === this.select });
+    }
+  }
+
+  /**
+   * Opens the dropdown.
+   */
+  openDropdown() {
+    if (this.state.isOpen || this.props.disabled) {
+      return;
+    }
+
+    if (this.input) {
+      this.input.focus();
+    }
+
+    this.setState({ isOpen: true, isFocused: true, isPositioned: false });
+  }
+
+  /**
+   * Positions the dropdown to utilize the most available space.
+   */
+  positionDropdown() {
+    if (!this.state.isOpen) {
+      return;
+    }
+
+    const { dropdownAttrs } = this.props;
+    const { select, dropdown, state } = this;
+    this.setState(Util.dropdownPosition(dropdownAttrs, state, select, dropdown));
+  }
+
+  /**
+   * Handles the blur event.
+   */
+  handleBlur(event) {
+    if (!this.state.excludeBlur) {
+      this.closeDropdown();
+    }
+
+    if (this.props.onBlur) {
+      this.props.onBlur(event);
+    }
+  }
+  /**
+   * Handles the focus event.
+   */
+  handleFocus(event) {
+    // Show results once input is focused
+    if (!this.state.showResults) {
+      this.setState({ showResults: true, isOpen: true });
```

Comment:
This might be challenging or unavoidable but this function can call setState twice and do a double render. Once here and once again if focus needs to be set. If possible setState should only be called once in a function